service/storage_proxy.cc

```diff
@@ -2141,16 +2141,24 @@ gms::inet_address storage_proxy::find_leader_for_counter_update(const mutation&
         throw exceptions::unavailable_exception(cl, block_for(ks, cl), 0);
     }
 
+    const auto my_address = utils::fb_utilities::get_broadcast_address();
+    bool i_can_be_leader = false;
     auto local_endpoints = boost::copy_range<std::vector<gms::inet_address>>(live_endpoints | boost::adaptors::filtered([&] (auto&& ep) {
+        i_can_be_leader |= ep == my_address;
```

Comment:
I personally do not like abusing copy_range like that. I think it should not be a problem to have additional pass over local_endpoints to look for my_address. But I know that some performance paranoid people may not agree with that.

Reply:
~Done. Most clients are token-aware (I think) so I put the short circuit even before the creation of `local_endpoints`.

Reply:
Ah, so you did it deliberately? Well, use boost::find then and add a comment that you do it deliberately. Or just drop the use of `boost::copy_range` and open code the loop that will return immediately if local address is found.

Reply:
Done. I chose the "early `boost::find` path" + comment.